### PR TITLE
Implement snug mode for the admin interface

### PR DESCRIPTION
**Pull request by thibaudcolas**
Implements #11265. I chose to introduce a global "UI density multiplier" as `--w-density-factor`, so we could introduce more options in the future, and make it easier for people wanting to tweak the UI density further. There will also be cases where the changes are better suited as theme overrides, with a global "theme" class, but I think those should be the exception.

Specific changes here:

- Spacing between panels from 40 to 25px (note the spacing between fields is at play here too, currently a static 25px)
- Spacing between field widgets and their label / help text from 10 to 5px.
- Title field font size
- Image chooser thumbnail size

From the designs I’ve left out:

- Fields height – this feels like a lot of risk of alignment issues for a small gain.
- Panel label font styles – same as above.
- Panel close "placeholder" icon (solid vs. outline) – the outline icon isn’t available in our FontAwesome license as far as I can see (assuming it’s [circle chevron down – outline](https://fontawesome.com/icons/circle-chevron-down?f=classic&s=solid))

## Documentation

I added minimal docs to explain how people can customize this for their site, and also how future UI components in Wagtail core or outside can leverage the new `--w-density-factor`. Decided not to be too explicit about other possible uses of the `w-density-snug` class to do more advanced overrides as I think we should keep those to a minimum.

## Testing this

I’d recommend testing with the bakerydemo homepage, in the "edit" view and listing view.

-   [x] Do the tests still pass?[^1]
-   [x] Does the code comply with the style guide?
-   ~~[ ] For Python changes: Have you added tests to cover the new/fixed behaviour?~~
-   [ ] For front-end changes: Did you test on all of Wagtail’s supported environments?[^2]
    -   [x] **Please list the exact browser and operating system versions you tested**: Chrome 123, Firefox 124, macOS 14.4
    -   [x] **Please list which assistive technologies [^3] you tested**: None
-   [x] For new features: Has the documentation been updated accordingly?

### CODE REVIEW

File: client/scss/components/_chooser.scss [hunk 1]
```diff
@@ -61,8 +63,20 @@ $preview-size: 2.625rem; // 42px
...
+  max-width: clamp(
+    math.div($max-width, 2),
+    calc($max-width * var(--w-density-factor)),
+    $max-width
+  );
+  max-height: clamp(
+    math.div($max-height, 2),
+    calc($max-height * var(--w-density-factor)),
+    $max-height
+  );
```
Comment (thibaudcolas) [inline client/scss/components/_chooser.scss:79]: The designed "snug" sizes are a bit different but I’d rather get the designs updated to use the exact same aspect ratio at all possible UI densities.

File: client/scss/components/_listing.scss [hunk 2]
```diff
@@ -27,10 +27,10 @@ ul.listing {
 
   td,
   th {
-    padding: 1.2em 0.3em;
+    padding: calc(1em * var(--w-density-factor)) 0.3em;
```
Comment (thibaudcolas) [inline client/scss/components/_listing.scss:30]: This and below – is a loose interpretation of the designs, as they’re quite out of sync with the build.

File: client/scss/components/_listing.scss [hunk 3]
```diff
@@ -48,16 +48,11 @@ ul.listing {
...
+    vertical-align: middle;
   }
 
   td.title {
     line-height: 2em;
-    padding: 1em 0.3em;
-
-    @include media-breakpoint-up(sm) {
-      padding: 1em;
-    }
```
Comment (thibaudcolas) [inline client/scss/components/_listing.scss:60]: This isn’t needed with `vertical-align: middle` for all cells.

File: client/scss/components/_listing.scss [hunk 4]
```diff
@@ -248,7 +243,7 @@ ul.listing {
       }
     }
 
-    .icon.initial {
+    .icon-folder {
```
Comment (thibaudcolas) [inline client/scss/components/_listing.scss:246]: Improves the alignment of the "lock" icon.

File: client/scss/components/forms/_title.scss [hunk 5]
```diff
@@ -1,19 +1,15 @@
 // Styles for title fields throughout the page.
 // The `title` class is meant to be applied by end users throughout different forms APIs.
-.w-panel.title,
-.w-field.title {
-  input,
-  textarea,
-  .Draftail-Editor .public-DraftEditor-content {
+:where(.w-panel, .w-field).title {
+  :where(input, textarea, .Draftail-Editor .public-DraftEditor-content) {
```
Comment (thibaudcolas) [inline client/scss/components/forms/_title.scss:4]: Here and below – refactoring the selectors so they’re easier to work with, should have no incidence on the styles.

**Review by laymonage — APPROVED** [review]

**Pull request merged**